src/test/java/com/epam/university/java/core/task046/Task046Test.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.university.java.core.task046;
+
+import com.epam.university.java.core.helper.TestHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class Task046Test {
+    private Task046 instance;
+
+    @Before
+    public void setUp() throws Exception {
+        instance = TestHelper.getInstance(Task046.class);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void calculateWithZeroSize() throws Exception {
+        instance.calculate(0, new HashMap<>());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void calculateWithNull() throws Exception {
+        instance.calculate(1,null);
+    }
+
+    @Test
+    public void calculate1() {
+        final Double expected = 180.000;
+        final Double result = instance.calculate(50,
+                new HashMap<>(Map.of(
```

Comment:
Why do you create `HashMap` here? Isn't it possible using just `Map.of()` here?

Reply:
Just forgot about this new feature of Java 11.

Reply:
No, surprisingly, you're using it because of `new HashMap<>(Map.of())`. I suppose, there is no need to wrap result of `Map.of()` into a new `HashMap`.

Reply:
Oh, well, got it. I'll fix it by this evening.

Reply:
Great, thanks.

Reply:
Fixed